src/openassetio-core/include/openassetio/trait/TraitBase.hpp

```diff
@@ -108,6 +108,24 @@ struct TraitBase {
    **/
   [[nodiscard]] bool isValid() const { return specification_->hasTrait(Derived::kId); }
 
+  /**
+   * Applies this trait to the specification.
+   *
+   * If the specification already has this trait, it is a no-op.
+   **/
+  void imbue() const { specification_->addTrait(Derived::kId); }
```

Comment:
[plan] We're still in a bit of a pickle here as this code isn't used anywhere at the mo (see the large disclaimer at the top of this file). We should figure out exactly what we want to do with autogen, and how we want the base classes to work.

Reply:
This is covered in #373 [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/373/commits/f384ee5929fde4017552bd5bb3097aefb83f239e).